modules/hal_nordic/nrfs/backends/nrfs_backend_ipc_service.c

```diff
@@ -120,14 +127,44 @@ static void ipc_sysctrl_ept_recv(const void *data, size_t size, void *priv)
 	}
 }
 
+/**
+ * @brief This function will try to send data directly using ipc service
+ *        In case of errors it will retry if configured.
+ *
+ * @param message Pointer to the buffer to send.
+ * @param size Number of bytes to send.
+ * @return see function @ref ipc_service_send
+ */
+static int nrfs_backend_try_send_directly_over_ipc_service(void *message, size_t size)
+{
+	size_t retry_count = CONFIG_NRFS_SEND_RETRY_MAX_COUNT;
+	int ret = ipc_service_send(&ipc_cpusys_channel_config.ipc_ept, message, size);
+
+	while (retry_count--) {
+		if (ret < (int)size) {
+			k_usleep(CONFIG_NRFS_SEND_RETRY_DELAY);
+			ret = ipc_service_send(&ipc_cpusys_channel_config.ipc_ept, message, size);
```

Comment:
Couldn't this be refactored to avoid double checking if `ret < (int)size` (here and outside the function) and calling `ipc_service_send()` in two places?